### ISSUE TITLE
Direct editing of python code within tactic!

### ISSUE THREAD

**Issue by bsherin** [bsherin/tactic#25]
It should be possible to do this. Code run for a tile can be accessed from the back of a tile, perhaps it will be shown in a dialog box.

Perhaps it will be broken up according to the different methods/handlers within a tile.

**Comment by arthurhjorth** [comment]
I think it's a cool idea, and it should be possible with the [refletion](https://docs.python.org/2/c-api/reflection.html) module for finding function signatures and [inspect](https://docs.python.org/2/library/inspect.html) module for retrieving the code. They wouldn't let you write your edited code back to the .py files, though.  So you would have to copy/paste the code back into the .py file, and then upload them as tiles and overwrite the one you edited the code in on the server. Or find a way of creating new tiles from the existing tile code + overwriting where you have added new code. That sounds tricky to me, but maybe there's a good hack for it.

Is your use case faster testing on the server? If so, maybe an easier solution would be to just have a Tile, the only function of which is to allow you to run code from a built-in code editor. It could contain a text input field for each Tile API function, and just evaluate whatever you write there when those functions are called. That way you could sort of do the same thing, but without having to implement the reflection/inspect stuff on all tiles?

**Comment by bsherin** [comment]
I've discovered a version of code editing that is pretty easy, and that I've got just about done. For user uploaded modules, we have the module stored as a string. So I'm letting the user edit that from their resource management page and then re load the module. I found javascript library, called CodeMirror, that makes the editing easy, with syntax highlighting and everything. I'm figuring we can also make available to the user on their resource page a set of standard modules they can edit. 

I *would* still like to allow editing right from a tile. A way to avoid some of the complicated inspect stuff would be to keep the code for tiles around as strings. Then we can use the same machinery I've already got up. The one complicated bit is that modules can contain the code for multiple tiles, and we'd like to only have the user see the code for the tile they're working on.
